android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java

```diff
@@ -168,4 +189,242 @@ private void ensureInitialized() {
             }
         }
     }
+
+    @Override
+    public JsonElement get(String key) {
+        ensureInitialized();
+
+        if (key.equals("confirmations") || key.equals("contaminations")) {
+            LOGGER.error("[Deprecation] Retrieving confirmations and contaminations from the Evolv context with keys \"confirmations\"," +
+                    "and \"contaminations\" is deprecated. Please use \"experiments.confirmations\" and \"experiments.contaminations\" instead.");
+        }
+// TODO: 27.07.2021 question: JS SDK - "return objects.getValueForKey(key, remoteContext) || objects.getValueForKey(key, localContext);"
+//  need to understand which cases should return value local or remote (if local and remote contexts aren't emprty)
+
+        if(localContext.size() != 0){
+            return helper.getValueForKey(key, localContext).getAsJsonObject();
+        }
+
+
+        JsonElement element = helper.getValueForKey(key, remoteContext);
+        return element;
+
+    }
+
+    // TODO: 27.07.2021 need to test!
+    @Override
+    public boolean pushToArray(String key, String value, boolean local) {
+        int limit = DEFAULT_QUEUE_LIMIT;
+
+        ensureInitialized();
+        JsonObject context = local ? localContext : remoteContext;
+
+        JsonArray newArray = new JsonArray();
+        JsonObject element = new JsonObject();
+
+        JsonElement originalArray = helper.getValueForKey(key, context);
+
+        if(originalArray == null){
+            LOGGER.error("The " + "\"" + key + "\"" +  " does not have a suitable value ");
+            return false;
+        }else if(originalArray.isJsonPrimitive()){
+            element.addProperty(key,value);
+        }else if(originalArray.isJsonObject()){
+            element.add(key,originalArray);
+        }
+
+        newArray.add(element);
+
+        for (int i = 0; i < newArray.size() ; i++) {
+            if(i > limit){
+                newArray.remove(i);
+            }
+        }
+        return set(key,newArray,local);
+    }
+
+    @Override
+    public boolean pushToArray(String key, String value, boolean local, int limit) {
+        ensureInitialized();
+        JsonObject context = local ? localContext : remoteContext;
+
+        JsonArray newArray = new JsonArray();
+        JsonObject element = new JsonObject();
+
+        JsonElement originalArray = helper.getValueForKey(key, context);
+
+        if(originalArray == null){
+            LOGGER.error("The " + "\"" + key + "\"" +  " does not have a suitable value ");
+            return false;
+        }else if(originalArray.isJsonPrimitive()){
+            element.addProperty(key,value);
+        }else if(originalArray.isJsonObject()){
+            element.add(key,originalArray);
+        }
+
+        newArray.add(element);
+
+        for (int i = 0; i < newArray.size() ; i++) {
```

Comment:
Just don't add the element if it's over the limit -- adding and removing seems wasteful

Reply:
ok i will redo